extensions/ql-vscode/src/query-history.ts

```diff
@@ -727,20 +737,20 @@ export class QueryHistoryManager extends DisposableObject {
       return;
     }
 
-    let p: string | undefined;
+    let externalFilePath: string | undefined;
     if (finalSingleItem.t === 'local') {
       if (finalSingleItem.completedQuery) {
-        p = finalSingleItem.completedQuery.query.querySaveDir;
+        externalFilePath = path.join(finalSingleItem.completedQuery.query.querySaveDir, 'timestamp');
       }
     } else if (finalSingleItem.t === 'remote') {
-      p = path.join(this.queryStorageDir, finalSingleItem.queryId);
+      externalFilePath = path.join(this.queryStorageDir, finalSingleItem.queryId, 'timestamp');
     }
 
-    if (p) {
+    if (externalFilePath) {
```

Comment:
This is probably a fringe case, but is it worth also throwing a `Failed to open` error when this path doesn't exist (e.g. if a user has deleted the timestamp for some reason)?
At the moment it fails silently.

Reply:
Hmmm...unfortunately, the `revealFileInOS` command does not throw an error if the file doesn't exist. I guess I can add a `fs.pathExists` check before trying to open.

Reply:
Done